spec/OpenLineage.yml

```diff
@@ -0,0 +1,250 @@
+        facets:
+          description: The facets for this dataset
+          type: object
+          properties:
+            schema:
+              $ref: '#/components/schemas/SchemaDatasetFacet'
+            dataSource:
+              $ref: '#/components/schemas/DatasourceDatasetFacet'
+          additionalProperties:
+            $ref: '#/components/schemas/CustomFacet'
+      required:
+        - datasetId
+
+    Origin:
+      description: Origin producer of this facet
+      type: object
+      properties:
+        name:
+          description: name of the project producing this metadata
+          type: string
+        version:
+          description: version of the project producing this metadata
+          type: string
+      required:
+       - name
+
+    JobId:
+      description: The identifier of a job # TODO: should this be a URN?
+      type: object
+      properties:
+        namespace:
+          description: The namespace containing that job
+          type: string
+          example: "my-scheduler-namespace"
+        name:
+          description: The unique name for that job within that namespace
+          type: string
+          example: "myjob.mytask"
+      required:
+       - namespace
+       - name
+
+    DatasetId:
+      description: The identifier of a dataset # TODO: should this be a URN?
+      type: object
+      properties:
+        namespace:
+          description: The namespace containing that dataset
+          type: string
+          example: "my-datasource-namespace"
+        name:
+          description: The unique name for that dataset within that namespace
+          type: string
+          example: "instance.schema.table"
+      required:
+       - namespace
+       - name
+
+# Custom facets
+    CustomFacet:
+      type: object
+      properties:
+        origin:
+          description: The name of the origin of this custom facet.
+          type: string
+        version:
+          description: The version of this custom facet.
+          type: string
+      additionalProperties: true
+      required:
+        - origin
+
+## Run Facets
+    NominalTimeRunFacet:
+      type: object
+      properties:
+        nominalStartTime:
+          description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal start time of the run. AKA the schedule time
+          type: string
+          format: date-time
+        nominalEndTime:
+          description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal end time of the run.
+          type: string
+          format: date-time
+
+    ParentRunFacet:
+      description: the id of the parent run and job, iff this run was spawn from an other run (for example, the Dag run scheduling its tasks)
+      type: object
+      properties:
+        runId:
+          type: string
+          format: uuid
+        jobId:
+          $ref: '#/components/schemas/JobId'
```

Comment:
This should be able to take an entire job entity rather than just the id components. This will allow it to be completely defined in a single event.

Reply:
I am intending that the Run in question has started before that one and so we could have 2 RunStateUpdates at the top level. One for each job run.
```
POST /events [
  { runID: "1", jobId: { ns: "foo", name: "parent"} ... },
  { runID: "2", jobId: { ns: "foo", name: "child"}, parent: {runID: "1", jobId: { ns: "foo", name: "parent"}} ... }
]
```